source/includes/read/cursors.kt

```diff
@@ -0,0 +1,65 @@
+package org.example
+import com.mongodb.ConnectionString
+import com.mongodb.CursorType
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.kotlin.client.MongoClient
+import org.bson.codecs.pojo.annotations.BsonId
+import org.bson.types.ObjectId
+
+// start-data-class
+data class Restaurant(
+    @BsonId
+    val id: ObjectId,
+    val name: String
+)
+// end-data-class
+
+fun main() {
+    val uri = "<connection string URI>"
+
+    val settings = MongoClientSettings.builder()
+        .applyConnectionString(ConnectionString(uri))
+        .retryWrites(true)
+        .build()
+
+    val mongoClient = MongoClient.create(settings)
+    val database = mongoClient.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+
+    // start-cursor-iterate
+   val results = collection.find()
+
+   results.forEach { result ->
+       println(results)
+   }
+    // end-cursor-iterate
+
+    // start-cursor-next
+   val results = collection.find<Restaurant>(eq(Restaurant::name.name, "Dunkin' Donuts"))
+
+   println(results.cursor().next())
+    // end-cursor-next
```

Comment:
It's not good that we are neither exhausting the cursor nor closing it explicitly here. Moreover, I think all our examples should be written such that they close the cursor via the `use` function (except for the "Retrieve All Documents" example, because it does not use a cursor). Having any example that does not guarantee the cursor to be closed is potentially detrimental, as users may just copy and paste it.

If we do this, then we won't need the "Close a Cursor" section. We still may leave a note somewhere saying that the driver closes exhausted cursors automatically.

Reply:
I can update the examples to include an explicit cursor close where necessary.

Question: When the example explicitly loops through the entire cursor (as in the iterative example) do we still need to include a manual cursor close given that we're exhausting the cursor by looping through the whole thing?